lib/ansible/plugins/action/net_template.py

```diff
@@ -93,6 +93,13 @@ def _handle_template(self):
         except IOError:
             return dict(failed=True, msg='unable to load src file')
 
+        searchpath = [working_path, os.path.dirname(source)]
+        if self._task._role is not None:
+            if self._task._block._dep_chain:
+                for role in sorted(self._task._block.get_dep_chain()):
+                    searchpath.insert(1, role._role_path)
+            searchpath.insert(1, self._task._role._role_path)
```

Comment:
Thank you @Qalthos and I agree with that suggestion.

I've incorporated your suggestion and run it locally, without any issue.

Please take a look at the updated change and let me know if you see any other issues.

Thank you for your support and hope we'll get the green sign from travis. :)

Cheers,